### PR TITLE
Fix memory leak in write_riemann

### PR DESCRIPTION
**Pull request by faxm0dem**
Changelog: Fix memory leak in write_riemann
Change-Id: If95da8d90c078c19ac0062e3bbb006185dcc441d

### CODE REVIEW

File: src/write_riemann.c [hunk 1]
```diff
@@ -641,6 +642,7 @@ static int wrr_config_node(oconfig_item_t *ci) /* {{{ */
 
   if ((host = calloc(1, sizeof(*host))) == NULL) {
     ERROR("write_riemann plugin: calloc failed.");
+    free(host);
```
Comment (rubenk) [inline src/write_riemann.c]: this has no effect

Comment (faxm0dem) [reply src/write_riemann.c]: I just realized that, thanks :-)

**Comment by faxm0dem** [comment]
I was trying to fix this (lesser) one:

```
==30694== 331 (192 direct, 139 indirect) bytes in 1 blocks are definitely lost in loss record 40 of 53
==30694==    at 0x4C2BFB9: calloc (vg_replace_malloc.c:762)
==30694==    by 0x6539FB6: wrr_config_node (write_riemann.c:643)
==30694==    by 0x6539FB6: wrr_config (write_riemann.c:853)
==30694==    by 0x1101BE: dispatch_block_plugin (configfile.c:408)
==30694==    by 0x1119E7: dispatch_block (configfile.c:442)
==30694==    by 0x1119E7: cf_read (configfile.c:1034)
==30694==    by 0x10F335: configure_collectd (collectd.c:360)
==30694==    by 0x10F335: init_config (collectd.c:410)
==30694==    by 0x10E6A0: main (cmd.c:167)
```

**Comment by faxm0dem** [comment]
But I'll do that in a separate PR. this one is major an needs to be merged asap ;-)

**Comment by rubenk** [comment]
I take it this doesn't just happen in master? Can you open this PR against the oldest stable branch where this applies?

**Comment by rpv-tomsk** [comment]
As I see, if such PR would be correct, untrusted maintainers still has no way to merge PR untill "code owner" from Intel approve this.

Be happy, dear maintainers, you treated as unskilled workers here.

**Comment by rubenk** [comment]
@faxm0dem can you rebase on that branch?

**Review by smortex — APPROVED** [review]
Ooops, looks like I introduced this bug in #3158. Thanks for fixing it!

**Review by smortex — APPROVED** [review]
Ooops, looks like I introduced this bug in #3158. Thanks for fixing it!

**Comment by faxm0dem** [comment]
@smortex read the commit message to know how much money you owe us 🤣

**Comment by smortex** [comment]
Outch! I would be interested in more details!

**Pull request merged**